packages/remix-cloudflare-workers/worker.ts

```diff
@@ -108,12 +117,21 @@ export function createEventHandler({
   });
 
   const handleEvent = async (event: FetchEvent) => {
-    let response = await handleAsset(event, build);
+    const cache = await caches.default;
+    let response: Response | null | undefined = await cache.match(
+      event.request
+    );
+    if (response && process.env.NODE_ENV !== "development") return response;
+
+    response = await handleAsset(event, build);
 
     if (!response) {
       response = await handleRequest(event);
     }
 
+    if (process.env.NODE_ENV !== "development")
+      event.waitUntil(cache.put(event.request, response.clone()));
```

Comment:
I had a question about the `cache.put` here. It has to do with streaming responses.

When Remix eventually supports streaming the response, we may not actually know if there are errors or not when we first start the stream. For example, we could start rendering a component before we have all the data for the child components, and one of those data fetchers could be slow or eventually fail. In this case, we may have already started flushing an HTML document to the browser but we would not be able to finish it, so we wouldn't want that result cached.

My question is: does the `cache.put` work differently here for streaming responses? Or would we need some conditional logic here to avoid caching the response if the body is a stream (and could potentially have errors)?

Reply:
`cache.put` will just blindly put whatever you give it into the cache.

When you come to implement streaming responses, you'd have to, within the waitUntil, consume the stream and decide whether or not to cache.

Something like this, I think:

```ts
event.waitUntil(response.clone().text().then(text => {
  if (responseIsOkay(text)) {
    return cache.put(event.request, response.clone())
  }
}))

return return
```

Reply:
That's what I expected. Thanks for confirming!